source/code/plugins/out_oms_changetracking_file.rb

```diff
@@ -0,0 +1,591 @@
+        @@PrimaryContentLocation = primaryContentLocation
+    end 
+    def get_SecondryContentLocation
+       return @@SecondryContentLocation
+    end 
+    def set_SecondryContentLocation(secondryContentLocation)
+        @@SecondryContentLocation = secondryContentLocation
+    end 
+
+    def configure(conf)
+      s = conf.add_element("secondary")
+      s["type"] = ChunkErrorHandler::SecondaryName
+
+      super
+      if !@PrimaryContentLocation.nil? and !@PrimaryContentLocation.empty? and @PrimaryContentLocation.include? "http" or @PrimaryContentLocation.include? "https"
+         urlDetails = @PrimaryContentLocation.split('?')
+         if !urlDetails.nil? and urlDetails.length == 2
+            @@ContentlocationUri = urlDetails[0].strip
+            @@PrimaryContentLocationAccessToken = urlDetails[1]
+         end
+      end
+      if !@SecondryContentLocation.nil? and !@SecondryContentLocation.empty? and @SecondryContentLocation.include? "http" or @SecondryContentLocation.include? "https"
+         urlDetails = @SecondryContentLocation.split('?')
+         if !urlDetails.nil? and urlDetails.length == 2
+            @@SecondryContentLocationAccessToken = urlDetails[1].strip
+         end
+      end
+    end
+
+    def start
+      super
+      @proxy_config = OMS::Configuration.get_proxy_config(@proxy_conf_path)
+      @sha256 = Digest::SHA256.new
+      @log.debug "buffer_path : #{@buffer_path}"
+      if !@buffer_path.empty?
+         contentlocationfilepath = File.dirname(@buffer_path) + '/' + @@ContentLocationCacheFileName
+
+         @log.debug "contentlocationfilepath : #{contentlocationfilepath}"
+         if File.exists?(contentlocationfilepath)
+            @@LastContentlocationUri = File.open(contentlocationfilepath, &:gets).strip
+         end
+      end
+      @log.debug "LastContentlocationUri : #{@@LastContentlocationUri}"
+    end
+
+    def shutdown
+      if !@buffer_path.empty?
+         contentlocationfilepath = File.dirname(@buffer_path) + '/' + @@ContentLocationCacheFileName
+         if File.exists?(contentlocationfilepath)
+            File.open(contentlocationfilepath, "w") do |f|
+                f.puts "#{@@ContentlocationUri}"
+            end 
+         else
+            File.write(contentlocationfilepath, "#{@@ContentlocationUri}")
+         end
+      end
+      @log.debug "LastContentlocationUri written to : #{contentlocationfilepath}"
+      super
+    end
+
+    ####################################################################################################
+    # Methods
+    ####################################################################################################
+
+    # create a HTTP request to PUT blob
+    # parameters:
+    #   uri: URI. blob URI
+    #   msg: string. body of the request
+    #   file_path: string. file path
+    # returns:
+    #   HTTPRequest. blob PUT request
+    def create_blob_put_request(uri, msg, request_id, file_path = nil)
```

Comment:
Please explain how this method is different from [out_oms_blob create_blob_get_request](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/source/code/plugins/out_oms_blob.rb#L57) and why it is necessary to make this a separate plugin from out_oms_blob. It would be much better if you utilized out_oms_blob entirely or merely its methods.